addons/enemybar/actionTracking.lua

```diff
@@ -0,0 +1,299 @@
+res = require('resources')
+vector = require('vectors')
+packets = require('packets')
+
+tracked_actions = {}
+tracked_enmity = {}
+tracked_debuff = {}
+framerate = 75
+clean_actions_delay = framerate
+clean_actions_tick = clean_actions_delay
+
+function looking_at(a, b)
+	local h = a.facing % math.pi
+	local h2 = (math.atan2(a.x-b.x,a.y-b.y) + math.pi/2) % math.pi
+	return math.abs(h-h2) < 0.15
+end
+
+function is_party_member_or_pet(mob)
```

Comment:
This function is rather slow (through little fault of your own). Checking on every action for every target, having to fetch entity objects through the already slow API (and occasionally more than once) and comparing many values for each entity on every action packet may be fast on your PC, but I would not be surprised if this would slow down other player's computers significantly.

There are a number of ways around that, most helpfully though, caching and removing some of the unnecessary checks. You can assume that any function in `windower.ffxi.` runs very poorly, especially `windower.ffxi.get_items`, `windower.ffxi.get_party` and `windower.ffxi.get_player`, two of which you use here. The player ID could be cached throughout the addon for example, as it only changes during logout/login (at which point you can update it).

You could also do some pre-filtering of the ID. For it to be a player ID it must be `id < 0x01000000`, and for it to be a pet ID it must be `id > 0x01000000 && id % 0x1000 > 0x700`. If it is neither, the function may return `false` safely.

But here you can use caching as well, and it should be pretty simple, since you only need to capture the party update packet (0x0DD) as well as the pet info packet (0x067) and build a simple dictionary from there. That would probably be the best solution for the given effort (not much).